plans/dht/test/utils.go

```diff
@@ -0,0 +1,36 @@
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/libp2p/go-libp2p-core/routing"
+)
+
+func monitorEvents(ctx context.Context, runenv *runtime.RunEnv) (context.Context, error) {
+	ctx, events := routing.RegisterForQueryEvents(ctx)
+	f, err := os.OpenFile(runenv.TestRun+".json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer f.Close()
+
+		for e := range events {
+			bytes, err := json.Marshal(e)
+			if err != nil {
+				panic(err)
```

Comment:
Slightly off-topic, but are `testplans` allowed to `panic`? If we allow `testplans` to panic, we should probably stop using the `Job` controller on k8s and switch to individual `Pods` - there is no way to tell k8s to not restart a failing pod that belongs to a job, and that's what would happen if some of the testplan instances panic.

Reply:
Not sure. How would you change this to avoid panicking? The only option I see is ignoring the errors or creating a channel and returning a possible multierror in the end.

Reply:
If we decide that a testplan should not return an exit code other than 0, we then either catch the panics and `recover` or return errors, there are no other options AFAIK.

I am not saying we should do that, just noting that this impacts the overall flow, and we must probably change the logic either on the testplan side or on the runner side.